src/bootstrap/src/core/builder/tests.rs

```diff
@@ -1235,24 +1236,38 @@ fn any_debug() {
 /// The staging tests use insta for snapshot testing.
 /// See bootstrap's README on how to bless the snapshots.
 mod staging {
+    use crate::Build;
+    use crate::core::builder::Builder;
     use crate::core::builder::tests::{
         TEST_TRIPLE_1, configure, configure_with_args, render_steps, run_build,
     };
+    use crate::utils::tests::ConfigBuilder;
 
     #[test]
     fn build_compiler_stage_1() {
-        let mut cache = run_build(
-            &["compiler".into()],
-            configure_with_args(&["build", "--stage", "1"], &[TEST_TRIPLE_1], &[TEST_TRIPLE_1]),
-        );
-        let steps = cache.into_executed_steps();
-        insta::assert_snapshot!(render_steps(&steps), @r"
-        [build] rustc 0 <target1> -> std 0 <target1>
-        [build] llvm <target1>
-        [build] rustc 0 <target1> -> rustc 1 <target1>
-        [build] rustc 0 <target1> -> rustc 1 <target1>
+        insta::assert_snapshot!(
+            ConfigBuilder::build()
+                .path("compiler")
+                .stage(1)
+                .get_steps(), @r"
+        [build] rustc 0 <host> -> std 0 <host>
```

Comment:
Discussion: yes, I find this very helpful. There are some bootstrap `->` messages (not in snapshot tests) that has always confused me (e.g. I think rustdoc or sth shows "stage 2 -> stage 3" in some cases). We may want to spell out what "->" actually is intended to mean here (maybe in the rustc-dev-guide bootstrap testing section)?

Insofar as I can tell, here `->` means "X is used to build Y" or "X builds Y" right?

Reply:
So I just reused the syntax that bootstrap prints, but these snapshot test strings are constructed only for tests, we're not printing what bootstrap prints. In the tests, `->` means "X is used to build Y", yes. In bootstrap output, some of the `->` outputs don't make any sense, and we should clean them up. In particular when testing something, it doesn't really make sense to print `->`, I think.

Reply:
Right, that seems fine. Maybe in this file or elsewhere, leave a comment about "not to be confused with bootstrap messages `->` that may undergo further changed/fixes"?